src/custom/GoogleDLPRedactor.ts

```diff
@@ -126,8 +132,26 @@ export class GoogleDLPRedactor implements IAsyncRedactor {
   constructor(private opts: GoogleDLPRedactorOptions = {}) {
     this.dlpClient = new DLP.DlpServiceClient(this.opts.clientOptions);
   }
-
   async redactAsync(textToRedact: string): Promise<string> {
+    const maxContentSize = this.opts.maxContentSizeForBatch || MAX_DLP_CONTENT_LENGTH;
+
+    if (textToRedact.length > maxContentSize && !this.opts.disableAutoBatchWhenContentSizeExceedsLimit) {
+      const batchResults = [];
+      let batchStartIndex = 0;
+      while (batchStartIndex < textToRedact.length) {
+        const batchEndIndex = batchStartIndex + maxContentSize;
+        const batchText = textToRedact.substring(batchStartIndex, batchEndIndex);
+        batchResults.push(await this.doRedactAsync(batchText));
+        batchStartIndex = batchEndIndex;
+      }
```

Comment:
minor improvement potential: Currently you're redacting each batch sequentially. Maybe it'd be better (certainly faster) to add all batchResults to an array and then await the array via Promise.all instead of awaiting each result in the loop iteration.

Reply:
Yeah good idea.. I don't think there is any reason we can't do them in parallel. I'll update the PR